cmake/EthCheckCXXCompilerFlag.cmake

```
@@ -10,9 +10,7 @@ include(CheckCXXCompilerFlag)
 #
 function(eth_add_cxx_compiler_flag_if_supported FLAG)
   # Remove leading - or / from the flag name.
-  string(REGEX REPLACE "^[-/]" "" name ${FLAG})
-  # Deletes any ':' because it's invalid variable names.
-  string(REGEX REPLACE ":" "" name ${name})
+  string(REGEX REPLACE "^-|/" "" name ${FLAG})
```

Won't this regex just match any `/` inside the string, e.g. in paths? Tests are passing so I guess not but asking just in case.

```suggestion
  string(REGEX REPLACE "^(-|/)" "" name ${FLAG})
```

---

This will also, in general break https://github.com/ethereum/solidity/blob/eb77ed080a44c456caedde666e8af813685a16f4/cmake/EthCompilerSettings.cmake#L29

---

Or depends... ah well... all of this is a mess in general :-D...

---

But yeah, the current version in the PR will remove all ``/`` in ``${CMAKE_SOURCE_DIR}``, which will result in the macro prefix option to stop matching the actual source dir, leading to that ending up in the binaries again in the end, avoiding which is the whole point of the macro-prefix option thing. So that's not good at least...

---

In general I'm not sure if there are any or many compiler options that we pass in here and that start with a ``-`` that will end up being valid MSVC options, if they start with ``/`` instead...